doi/src/main/java/ca/nrc/cadc/doi/DoiAction.java

```diff
@@ -205,7 +284,271 @@ private void parsePath() {
                     includePublic = true;
                 }
             }
+        } else {
+            String requestPath = syncInput.getRequestPath();
+            String[] parts = requestPath.split("/");
+
+            if (parts.length > 0 && parts[2].equals("search")) {
+                doiAction = DoiAction.SEARCH_ACTION;
+                if (!syncInput.getParameterNames().isEmpty()) {
+                    prepareDoiStatusSearchFilter();
+                }
+            }
+        }
+    }
+
+    private void prepareDoiStatusSearchFilter() {
```

Comment:
This method could be moved to the PostAction, where you can do the check in parsePath() and create the search filter if necessary.